lib/virt_autotest/utils.pm

```diff
@@ -362,17 +363,31 @@ sub download_script {
     my ($script_name, %args) = @_;
     my $script_url = $args{script_url} // data_url("virt_autotest/$script_name");
     my $machine = $args{machine} // 'localhost';
+    $args{proceed_on_failure} //= 0;
 
     my $cmd = "curl -o ~/$script_name $script_url";
     $cmd = "ssh root\@$machine " . "\"$cmd\"" if ($machine ne 'localhost');
     unless (script_retry($cmd, timeout => 900, retry => 2, die => 0) == 0) {
         # Add debug codes as the url only exists in a dynamic openqa URL
         record_info("URL is not accessible", "$script_url", result => 'fail') unless head($script_url);
         unless ($machine eq 'localhost') {
-            record_info("machine is not ssh accessible", "$machine", result => 'fail') unless script_run("ssh root\@$machine 'hostname'") == 0;
-            record_info("OSD is unaccessible from $machine", "that means the machine is having problem to access SUSE network", result => 'fail') unless script_run("ssh root\@$machine 'ping -c3 openqa.suse.de'") == 0;
+            # Have to output debug info at here because no logs will be uploaded if there are connection problems
+            if (script_run("ssh root\@$machine 'hostname'") == 0) {
+                script_run("ssh root\@$machine 'ping -c3 openqa.suse.de'");
+                script_run("ssh root\@$machine 'nslookup " . get_var('WORKER_HOSTNAME', 'openqa.suse.de') . "'");
+                script_run("ssh root\@$machine 'cat /etc/resolv.conf'");
+            }
```

Comment:
Maybe you also need ```dnsdomainname``` if take the nature of the problem into consideration.

Reply:
For your download logic here, I would suggest using curl to check url availability in the first place before starting downloading. ```curl -I``` will also returns http status code.

Reply:
Even if the URL does not exist, curl can always get some content. fg. https://openqa.suse.de/tests/12816981#step/sriov_network_card_pci_passthrough/140

Reply:
> Maybe you also need `dnsdomainname` if take the nature of the problem into consideration.

added. thanks.

Reply:
You need to use ```curl -I```, for example:
```
waynechen:~ # curl -I http://does-not-exist.com
HTTP/1.1 302 Found
date: Fri, 17 Nov 2023 07:06:16 GMT
server: Apache
set-cookie: __tad=1700204776.8363663; expires=Mon, 14-Nov-2033 07:06:16 GMT; Max-Age=315360000
location: http://ww25.does-not-exist.com/?subid1=20231117-1806-1613-9bef-97bd32ee250b
content-type: text/html; charset=UTF-8
connection: close
```

Reply:
> For your download logic here, I would suggest using curl to check url availability in the first place before starting downloading. `curl -I` will also returns http status code.

The URL is checked on host via
`record_info("URL is not accessible", "$script_url", result => 'fail') unless head($script_url);`.

Do you mean move this line ahead before run curl inside the guest?
`unless (script_retry($cmd, timeout => 900, retry => 2, die => 0) == 0) {`

Reply:
> > Maybe you also need `dnsdomainname` if take the nature of the problem into consideration.
> 
> added. thanks.

dnsdomainname: `name or service not known` in guest, 
https://openqa.suse.de/tests/12849090#step/sriov_network_card_pci_passthrough/148

Reply:
OK.

Reply:
> > For your download logic here, I would suggest using curl to check url availability in the first place before starting downloading. `curl -I` will also returns http status code.
> 
> The URL is checked on host via `record_info("URL is not accessible", "$script_url", result => 'fail') unless head($script_url);`.
> 
> Do you mean move this line ahead before run curl inside the guest? `unless (script_retry($cmd, timeout => 900, retry => 2, die => 0) == 0) {`

Move the URL checking ahead, and verification run passed. http://openqa.suse.de/tests/12849138#step/sriov_network_card_pci_passthrough/138